grafana-plugin/src/containers/IntegrationTemplate/IntegrationTemplate.tsx

```diff
@@ -19,29 +19,42 @@ import TemplatePreview from 'containers/TemplatePreview/TemplatePreview';
 import TemplatesAlertGroupsList from 'containers/TemplatesAlertGroupsList/TemplatesAlertGroupsList';
 import { AlertReceiveChannel } from 'models/alert_receive_channel/alert_receive_channel.types';
 import { Alert } from 'models/alertgroup/alertgroup.types';
+import { ChannelFilter } from 'models/channel_filter/channel_filter.types';
+import LocationHelper from 'utils/LocationHelper';
 
 import styles from './IntegrationTemplate.module.css';
 
 const cx = cn.bind(styles);
 
 interface IntegrationTemplateProps {
   id: AlertReceiveChannel['id'];
+  channelFilterId?: ChannelFilter['id'];
   template: TemplateForEdit;
   templateBody: string;
   onHide: () => void;
   onUpdateTemplates: (values: any) => void;
-  onUpdateRoute: (values: any) => void;
+  onUpdateRoute: (values: any, channelFilterId?: ChannelFilter['id']) => void;
 }
 
 const IntegrationTemplate = observer((props: IntegrationTemplateProps) => {
-  const { id, onHide, template, onUpdateTemplates, onUpdateRoute, templateBody } = props;
+  const { id, onHide, template, onUpdateTemplates, onUpdateRoute, templateBody, channelFilterId } = props;
 
   const [isCheatSheetVisible, setIsCheatSheetVisible] = useState<boolean>(false);
   const [chatOps, setChatOps] = useState(undefined);
   const [alertGroupPayload, setAlertGroupPayload] = useState<JSON>(undefined);
   const [changedTemplateBody, setChangedTemplateBody] = useState<string>(templateBody);
   const [resultError, setResultError] = useState<string>(undefined);
 
+  if (template.isRoute) {
+  }
+
+  useEffect(() => {
+    LocationHelper.update({ template: template.name }, 'partial');
```

Comment:
These calls might be worthwhile doing it all in a batch when `template.isRoute` equals `true`. Something like this
```jsx
const locationParams: any = { template: template.name };
if (template.isRoute) {
    locationParams.routeId = channelFilterId;
}

LocationHelper.update(locationParams, 'partial');
```